src/overlays/actors/ovl_En_GirlA/z_en_girla.c

```diff
@@ -20,90 +62,544 @@ const ActorInit En_GirlA_InitVars = {
+      EnGirlA_BuyBottle, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SWORD_4, GID_SWORD_GREAT_FAIRY, func_800B8050, 4, 0X29F2, 0X29F3, GI_SWORD_GREAT_FAIRY,
+      EnGirlA_CanBuySword, EnGirlA_BuySword, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SWORD_1, GID_SWORD_KOKIRI, func_800B8050, 1, 0X29F4, 0X29F5, GI_SWORD_KOKIRI, EnGirlA_CanBuySword,
+      EnGirlA_BuySword, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SWORD_2, GID_SWORD_RAZOR, func_800B8050, 2, 0X29F4, 0X29F5, GI_SWORD_RAZOR, EnGirlA_CanBuySword,
+      EnGirlA_BuySword, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SWORD_3, GID_SWORD_GILDED, func_800B8050, 3, 0X29F4, 0X29F5, GI_SWORD_GILDED, EnGirlA_CanBuySword,
+      EnGirlA_BuySword, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SHIELD_2, GID_SHIELD_HERO, func_800B8050, 1, 0X29F6, 0X29F7, GI_SHIELD_HERO_2, EnGirlA_CanBuyShieldHero,
+      EnGirlA_BuyShieldHero, EnGirlA_BuyFanfare },
+    { OBJECT_GI_SHIELD_3, GID_SHIELD_MIRROR, func_800B8050, 1, 0X29F6, 0X29F7, GI_SHIELD_MIRROR,
+      EnGirlA_CanBuyShieldMirror, EnGirlA_BuyShieldMirror, EnGirlA_BuyFanfare },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_GirlA_0x80863870/func_808649C8.asm")
+void EnGirlA_SetupAction(EnGirlA* this, EnGirlAActionFunc action) {
+    this->actionFunc = action;
+}
+
+void EnGirlA_InitObjIndex(EnGirlA* this, GlobalContext* globalCtx) {
+    s16 params = this->actor.params;
+
+    //! @bug: Condition is impossible
+    if (params >= 43 && params < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->objIndex = Object_GetIndex(&globalCtx->objectCtx, sShopItemEntries[params].objectId);
+    if (this->objIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->actor.params = params;
+    this->mainActionFunc = EnGirlA_InitalUpdate;
+}
+
+void EnGirlA_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGirlA* this = THIS;
+
+    EnGirlA_InitObjIndex(this, globalCtx);
+}
+
+void EnGirlA_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 EnGirlA_CanBuyPotionRed(GlobalContext* globalCtx, EnGirlA* this) {
+    if (!func_80114E90()) {
+        return CANBUY_RESULT_NEED_EMPTY_BOTTLE;
+    }
+    if (gSaveContext.rupees < globalCtx->msgCtx.unk1206C) {
+        return CANBUY_RESULT_NEED_RUPEES;
+    }
+    return CANBUY_RESULT_SUCCESS;
+}
+
+s32 EnGirlA_CanBuyPotionGreen(GlobalContext* globalCtx, EnGirlA* this) {
+    if (!func_80114E90()) {
+        return CANBUY_RESULT_NEED_EMPTY_BOTTLE;
+    }
+    if (gSaveContext.rupees < globalCtx->msgCtx.unk1206C) {
+        return CANBUY_RESULT_NEED_RUPEES;
+    }
+    return CANBUY_RESULT_SUCCESS;
+}
+
+s32 EnGirlA_CanBuyPotionBlue(GlobalContext* globalCtx, EnGirlA* this) {
+    if (!(gSaveContext.weekEventReg[53] & 8)) {
+        return CANBUY_RESULT_CANT_GET_NOW;
+    }
+    if (!func_80114E90()) {
+        return CANBUY_RESULT_NEED_EMPTY_BOTTLE;
+    }
+    if (!(gSaveContext.weekEventReg[53] & 16)) {
```

Comment:
```suggestion
    if (!(gSaveContext.weekEventReg[53] & 0x10)) {
```
I think with bitflags (with the exception of single digits), it looks better to have it in hex